src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.ts

```diff
@@ -77,30 +89,65 @@ export class HostFormComponent extends CdForm implements OnInit {
     });
   }
 
+  // expand hostnames in case hostname is a pattern
+  checkHostNamePattern(hostname: string) {
+    const braceExpansion = require('brace-expansion');
+    if (hostname.includes('[') && hostname.includes(']') && !hostname.match(/(?![^(]*\)),/g)) {
+      const hostnameRange = hostname
+        .replace(/(?<=\d)\s*-\s*(?=\d)/g, '..')
+        .replace(/\(/g, '{')
+        .replace(/\)/g, '}')
+        .replace(/\[/g, '{')
+        .replace(/]/g, '}');
+      this.hostnameArray = braceExpansion(hostnameRange);
+    } else if (hostname.includes(',')) {
+      let hostArray = [];
+      hostArray = hostname.split(',');
+      hostArray.forEach((host: string) => {
+        if (host.includes('[') && host.includes(']')) {
+          const hostnameRange = host
+            .replace(/(?<=\d)\s*-\s*(?=\d)/g, '..')
+            .replace(/\(/g, '[')
+            .replace(/\)/g, ']')
+            .replace(/\[/g, '{')
+            .replace(/]/g, '}');
+          this.hostnameArray = this.hostnameArray.concat(braceExpansion(hostnameRange));
+        } else {
+          this.hostnameArray.push(host);
+        }
+      });
```

Comment:
BTW we can simply convert the comma-separated list to a brace expression by `{comma-separated-list}`, so perhaps we can simply enclose any input expressions in braces, and remove this else block assimilating it the first block. What do you think?